composer.json

```diff
@@ -26,24 +29,27 @@
         "symfony/filesystem": "^5.0.0 || ^6.0.0 || ^7.0.0",
         "symfony/finder": "^5.0.0 || ^6.0.0 || ^7.0.0",
         "symfony/translation": "^5.0.0 || ^6.0.0 || ^7.0.0",
-        "symfony/validator": "^5.0.0 || ^6.0.0 || ^7.0.0"
+        "symfony/validator": "^5.0.0 || ^6.0.0 || ^7.0.0",
+        "symfony/polyfill-php82": "^1.31",
+        "symfony/polyfill-php83": "^1.31",
+        "symfony/polyfill-php84": "^1.31"
     },
     "require-dev": {
-        "ext-pdo": "*",
-        "ext-json": "*",
-        "ext-xml": "*",
         "monolog/monolog": "^1.3 || ^2.3 || ^3.0",
         "phpstan/phpstan": "^1.2",
         "phpunit/phpunit": "^9.5.0",
         "spryker/code-sniffer": "^0.17.2",
-        "psalm/phar": "^5",
+        "psalm/phar": "^6",
         "mikey179/vfsstream": "^1.6"
     },
     "suggest": {
         "monolog/monolog": "The recommended logging library to use with Propel."
     },
     "replace": {
-        "propel/propel": "dev-main as 2.0.x-dev"
+        "propel/propel": "dev-main as 2.0.x-dev",
+        "symfony/polyfill-ctype": "*",
+        "symfony/polyfill-intl-grapheme": "*",
+        "symfony/polyfill-intl-normalizer": "*"
```

Comment:
These should have been placed in either `require` or `require-dev`, because this breaks when trying to install with a project that also specifies these replacements.

Replacements for the polyfills should not be specified by libraries, only by consumers.

Reply:
Right, yes, that makes sense. Thank you for reporting!

PR with fix is at #55, I just removed the lines in `replacement`. Let me know if you think there's a better way, otherwise I'll merge it right away.

Reply:
(and thanks for the quick follow-up!)

Reply:
It's my (everybody's) favorite kind of fix, just remove few lines and all works again :)

Those statements should avoid these libraries being loaded due to Propel/perpl, as we don't need them. It's suggested to do so on the symfony/polyfill page ([here](https://github.com/symfony/polyfill?tab=readme-ov-file#design)), but I could have realized that this means perpl declares there is no need to load those packages ever at all. Thank you for finding and reporting!

Fix is merged and should be delivered through `composer update`.
Cheers!